### ISSUE TITLE
lint deceiving formatting: if { .. } if { .. } 

### ISSUE THREAD

**Issue by llogiq** [Manishearth/rust-clippy#718]
If there are two consecutive if-expressions in a block and the second begins on the same line the first ends, the programmer may have wanted to do an `else if`, but forgot the `else`. Suggest to add either the `else` or a line break between the statements to reduce confusion. (Nicked from GCC6)

Example:
```
fn main() {
	let x = 0;
	if x == 0 {
		println!("Hello");
	} if x == 1 { //~ERROR please insert a line break or an `else`
		println!("World!");
	}
}
```

**Comment by mcarton** [comment]
Related: http://www.viva64.com/en/examples/V705/
If there is an `else if` with the `else` and `if` on different lines, it should warn as well:
```rust
if foo {
} else

if bar {
}
```

**Comment by llogiq** [comment]
Good idea! This should actually be quite easy to do, the hardest part is getting the span right (it may be better for reporting to construct a span *between* the statements, but we should of course check if the ExpnIds are equal.
